agent/xds/endpoints.go

```diff
@@ -155,6 +157,51 @@ func (s *Server) endpointsFromSnapshotMeshGateway(cfgSnap *proxycfg.ConfigSnapsh
 		resources = append(resources, la)
 	}
 
+	// generate the endpoints for the service subsets
+	for svc, resolver := range cfgSnap.MeshGateway.ServiceResolvers {
+		for subsetName, subset := range resolver.Subsets {
+			clusterName := ServiceSNI(svc, subsetName, "default", cfgSnap.Datacenter, cfgSnap)
+
+			endpoints := cfgSnap.MeshGateway.ServiceGroups[svc]
+
+			// locally execute the subsets filter
+			filterExp := subset.Filter
+			if subset.OnlyPassing {
```

Comment:
It's funny that you implemented this thing in the gateway side and I still had it listed as a TODO on my side without gateways because I hadn't gotten into the health endpoint yet.

Reply:
I may just steal this now seeing how easy this is to implement.